src/NzbDrone.Core/Music/TrackRepository.cs

```diff
@@ -37,24 +37,24 @@ public TrackRepository(IMainDatabase database, IEventAggregator eventAggregator,
             _logger = logger;
         }
 
-        public Track Find(int artistId, int albumId, int trackNumber)
+        public Track Find(string artistId, string albumId, int trackNumber)
         {
-            return Query.Where(s => s.ArtistId == artistId)
-                               .AndWhere(s => s.AlbumId == albumId)
+            return Query.Where(s => s.ForeignTrackId == artistId)
```

Comment:
We cant compare the remote Lidarr ID for the track to the artistId. They would not be the same. We need to compare the local artistId stored in the track table to the artistId provided. ForeignArtistId and ForeignAlbumId are not currently stored in the track table, only local DB ArtistId and AlbumId (the actual row id in local db).